src/Storefront/Resources/app/storefront/src/plugin/header/search-widget.plugin.js

```diff
@@ -85,33 +85,43 @@ export default class SearchWidgetPlugin extends Plugin {
      * @private
      */
     _handleInputEvent() {
-        const value = this._inputField.value.trim();
+        const formData = new FormData(this.el);
+        const inputValue = formData.get('search');
+        let encodedUri = '?';
 
         // stop search if minimum input value length has not been reached
-        if (value.length < this.options.searchWidgetMinChars) {
+        if (inputValue.length < this.options.searchWidgetMinChars) {
             // further clear possibly existing search results
             this._clearSuggestResults();
             return;
         }
 
-        this._suggest(value);
+        for(const [fieldName, fieldValue] of formData) {
+            if(encodedUri.charAt(encodedUri.length - 1) !== '?') {
+                encodedUri += '&';
+            }
+
+            encodedUri += `${fieldName}=` + encodeURIComponent(fieldValue);
+        }
+
+        this._suggest(encodedUri);
 
-        this.$emitter.publish('handleInputEvent', { value });
+        this.$emitter.publish('handleInputEvent', { inputValue, formData });
```

Comment:
this might brake plugins due to another name. You should be able to just write value:inputValue
this should also be documented (changelog.md -> handleInputEvent now emits... sth. along that way)